### ISSUE TITLE
`AccessibilityRenderExtension` doesn't work correctly with `RenderingMode.SHRINK`

### ISSUE THREAD

**Issue by VirtualParticle** [cashapp/paparazzi#1350]
**Description**
Using the `AccessibilityRenderExtension` is useful for analyzing accessibility information, but the information added to the side of the screenshot is cropped out when the `SHRINK` rendering mode is used.

**Steps to Reproduce**
Here is an example test:
```kotlin
class ExampleTest {

    @get:Rule
    val paparazzi = Paparazzi(
        renderingMode = RenderingMode.SHRINK,
        renderExtensions = setOf(AccessibilityRenderExtension()),
    )

    @Test
    fun exampleTest() {
        paparazzi.snapshot {
            Box(
                modifier = Modifier
                    .background(Color.White)
                    .padding(16.dp)
            ) {
                Text(text = "Hello, world!")
            }
        }
    }

}
```

**Expected behavior**
I would expect for the accessibility information on the side of the screenshot to be included in the cropped screenshot, that is to say that the crop would take into account the size of the accessibility information, even if it was taller than the actual content.

**Additional information:**
- Paparazzi Version: 1.3.3
- OS: MacOS
- Compile SDK: 34
- Gradle Version: 8.6
- Android Gradle Plugin Version: 8.3.0

**Screenshots**
Screenshot using `SHRINK`:
![image](https://github.com/cashapp/paparazzi/assets/20135416/c12ad84e-838e-434e-81eb-c35a594cdfaf)

Screenshot using `NORMAL`:
![image](https://github.com/cashapp/paparazzi/assets/20135416/0d845e92-63df-42ae-964d-419d309c7ba8)

I've given fixing this a look and it seems like one of the challenges here is that the accessibility information is a canvas element which can't easily use things like `WRAP_CONTENT`.

**Comment by MedetZhakupov** [comment]
Is it considered blocker for the release 1.3.4 @adamalyyan ?

**Comment by adamalyyan** [comment]
I don't think this would be considered a blocker but I'll defer to @geoff-powell as his PR will resolve this!

**Comment by geoff-powell** [comment]
Ya I think this might not make 1.3.4. Need to figure out a better way to render accessibility information in a layout so we can correctly measure the details view.

**Comment by colinmarsch** [comment]
I've been investigating the possible resolutions to this issue and I wonder if the best path forward might be to simply disallow accessibility snapshots from being taken in the `SHRINK` rendering mode.

It seems to me there are two possible technical paths to resolve this issue, both with drawbacks:
1. @geoff-powell 's PR (https://github.com/cashapp/paparazzi/pull/1355) can enable `SHRINK` support for accessibility snapshots, however some cases where the legend size is larger than the view under test will run into the legend text being cutoff
2. We find a way to ensure that the legend size can be properly measured (likely requiring another layout pass) and then base the generated snapshot size off of that. In this case the `SHRINK` rendering mode feels not respected as the generated snapshot will no longer be directly constrained to the size of the view under test.

Because of this, I think the option to disable the `SHRINK` rendering mode when the `AccessibilityRenderExtension` is used (with clear error messaging stating this) is potentially the best path forward here. If there are situations where it is vital to use `SHRINK` with the `AccessibilityRenderExtension` that I might be missing I would love to learn more about them!

**Comment by VirtualParticle** [comment]
The best use-case for `SHRINK` and accessibility is just testing smaller components that don't need a whole device-sized screenshot. Our solution has been to manually crop the size with `screenHeight` and `screenWidth` or just live with the larger screenshot.
